scripts/issues.coffee

```diff
@@ -0,0 +1,40 @@
+# Description:
+#   Create GitHub issues for CH beheer and PC.COM. Please be concise.
+#
+# Dependencies:
+#   octonode
+#
+# Configuration:
+#   HUBOT_GITHUB_TOKEN = <accessToken>
+#
+# Commands:
+#   @pccom <query> - Creates a new issue and tags it 'pccom'
+#   @beheer <query> - Creates a new issue and tags it 'beheer'
+
+github = require 'octonode'
+
+githubToken = process.env.HUBOT_GITHUB_TOKEN;
+
+module.exports = (robot) ->
+  unless githubToken?
+    robot.logger.error "Missing HUBOT_GITHUB_TOKEN in environment"
+    return
+  robot.listen(
+    (message) ->
+      console.log message
+      match = message.text.match /^@(pccom|beheer) (?:me )?(.*)/i
+      if match
+        data = {
+          title: match[2],
+          body: "Issue added by #{message.user.name}"
+          labels: ['slack', match[1]]
+        }
+        client = github.client githubToken
+        client.repo("WISVCH/pccom-beheer").issue data, (err) ->
+          if err
+            throw new Error "Github Error: #{err.message}"
```

Comment:
Kan, moet de data voor github even buiten de functie om worden bewaard.

On Sun, Mar 20, 2016, 17:15 Mark Janssen <notifications@github.com> wrote:

> In scripts/issues.coffee
> <https://github.com/WISVCH/florriebot/pull/99#discussion_r56766886>:
>
> > +    robot.logger.error "Missing HUBOT_GITHUB_TOKEN in environment"
> > +    return
> > +  robot.listen(
> > +    (message) ->
> > +      console.log message
> > +      match = message.text.match /^@(pccom|beheer) (?:me )?(.*)/i
> > +      if match
> > +        data = {
> > +          title: match[2],
> > +          body: "Issue added by #{message.user.name}"
> > +          labels: ['slack', match[1]]
> > +        }
> > +        client = github.client githubToken
> > +        client.repo("WISVCH/pccom-beheer").issue data, (err) ->
> > +          if err
> > +            throw new Error "Github Error: #{err.message}"
>
> De github issue code naar de (response) verplaatsen en dan pas
> response.reply doen als er daadwerkelijk een issue is aangemaakt?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/WISVCH/florriebot/pull/99/files/8e08278e80c724aa818392d85b1c949490fae498#r56766886>
>